### PR TITLE
feat(selectors): makeIsInitialRequestPendingSelector

### PR DESCRIPTION
**Pull request by BrentLayne**
# NOTE: 

Do not use this feature if possible.  Feature is subject to removal at any point in a future release.

### Why
`makeIsPendingSelector` will return `true` whenever the provided async action is pending.  This can be problematic when performing re-fetches (ex. after the action's TTL has expired), since we may only actually care about the pending status of the _initial_ request.  

This PR solves this issue by adding a `makeIsInitialRequestPendingSelector`, which returns `true` only if the action is `pending` and has never `completed`.

### What changed
- [x] In the reducer, set a `completed` flag on the provided action in response to `ASYNC_COMPLETE` events
  - [x] Persist the `completed` flag on subsequent pending events
- [x] Add a `makeIsInitialRequestPendingSelector` that checks whether the action is `pending` and has never `completed`

### How I tested
- [x] Update reducer tests
- [x] Add selector tests
- [x] Tested by symlinking my PR branch to an existing project

### CODE REVIEW

File: src/async.reducer.test.js [hunk 1]
```diff
@@ -36,45 +35,10 @@ describe('AsyncAction reducer', () => {
     const finalState = asyncActionReducer(newState, completeAction);
 
     expect(finalState).toEqual({
-      GET_FOOS_BY_NAME: {},
-    });
-  });
-
-  it('should record a pending request', () => {
```
Comment (BrentLayne) [inline src/async.reducer.test.js:43]: This test was a duplicate of the above test

**Comment by SethDavenport** [comment]
`pending` means the request is pending, by design.

The decision to show a spinner should depend on the **Presence or absence of data**.

See https://code.w10e.com/docs/sites/eng-handbook/engineering-guides/stack-guides/javascript/react-redux-common-patterns/#step-1-start-with-presentation_1

I'm concerned that by adding this selector, we're legitimizing a fundamentally flawed approach to making render time decisions.

**Render needs to be a pure function of the current state**. This is a fundamental attribute of functional reactive programming. Rendering different things based on current state plus knowledge of past actions is imperative thinking and doesn't fit with Redux as a model.

**Comment by SethDavenport** [comment]
That said, if we need this as a workaround to move forward, I'm ok adding it as long as it's accompanied by a big old comment saying why it shouldn't be used.

File: src/async.reducer.test.js [hunk 2]
```diff
@@ -3,77 +3,71 @@ import { asyncActionReducer } from './async.reducer';
...
-    const state = {};
-    const pendingAction = {
-      type: 'GET_FOOS_BY_NAME',
-      meta: {
-        status: 'ASYNC_PENDING',
-        identifier: 'nameOfTheFoo',
-      },
-    };
+  describe('a pending request', () => {
```
Comment (BrentLayne) [inline src/async.reducer.test.js:6]: Added a second pending test so I nested the tests under a new describe block ('a pending request')
The diff is easier to see [without whitespace](https://github.com/wealthsimple/async-action/pull/346/files?w=1)

File: .flowconfig [hunk 3]
```diff
@@ -13,6 +13,7 @@
 [libs]
 
 [options]
+esproposal.optional_chaining=enable
```
Comment (SethDavenport) [inline .flowconfig:16]: :100:

**Review by SethDavenport — APPROVED** [review]

**Pull request merged**